### PR TITLE
RA-1105 classAdd works now as classEdit

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
@rkorytkowski What do you think about that  ?

PS. don't merge this

### CODE REVIEW

File: app/components/classEdit/classEdit.html [hunk 1]
```diff
@@ -14,11 +14,11 @@
 						<textarea name="description" rows="3" cols="40" ng-model="vm.singleClass.description"></textarea>
 					</div>
 
-					<div>
+					<div ng-show="!vm.addingOption">
 						{{"CreatedBy" | translate}}:&#160;&#160;{{vm.singleClass.auditInfo.creator.display}} - {{vm.singleClass.auditInfo.dateCreated | date:'medium'}}
 					</div>
 
-					<div>
+					<div ng-show="!vm.addingOption">
```
Comment (rkorytkowski) [inline app/components/classEdit/classEdit.html]: I'd just use ng-show="vm.singleClass.uuid"

File: app/components/classEdit/classEdit.html [hunk 2]
```diff
@@ -14,11 +14,11 @@
 						<textarea name="description" rows="3" cols="40" ng-model="vm.singleClass.description"></textarea>
 					</div>
 
-					<div>
+					<div ng-show="!vm.addingOption">
```
Comment (rkorytkowski) [inline app/components/classEdit/classEdit.html]: Why not to simply use ng-show="vm.singleClass.auditInfo"?

File: app/components/conceptDictionaryApp.module.js [hunk 3]
```diff
@@ -25,9 +25,14 @@ function conceptDictionaryAppConfig($routeProvider, openmrsRest) {
...
 	  }).
 	  when('/class/add', {
-		  templateUrl: 'components/classAdd/classAdd.html',
-		  controller: 'ClassAddController',
-		  controllerAs: 'vm'
+		  templateUrl: 'components/classEdit/classEdit.html',
+		  controller: 'ClassEditController',
+		  controllerAs: 'vm',
+		  resolve: {
+			  singleClass : function(){
```
Comment (rkorytkowski) [inline app/components/conceptDictionaryApp.module.js]: Any particular reason you add it to resolve? I'd just declare it in controller.

File: app/components/classEdit/classEdit.controller.js [hunk 4]
```diff
@@ -36,18 +31,25 @@ angular
 		vm.editClass = editClass;
 		
 		vm.cancel = cancel;
+        
+        activate();
+        
+        function activate(){
+            if(singleClass.name == ''){
+                vm.addingOption = true;
```
Comment (rkorytkowski) [inline app/components/classEdit/classEdit.controller.js]: I don't think it's necessary at all, but if you want to have it then rename to vm.editMode.

**Comment by rkorytkowski** [comment]
I think it's the way to do it.

**Comment by AdamGrzybkowski** [comment]
@rkorytkowski I've added function to resolve, because we need to inject something into the controller

**Comment by rkorytkowski** [comment]
@AdamGrzybkowski, it looks good. Is it ready for merging?

**Comment by AdamGrzybkowski** [comment]
@rkorytkowski Tests are passing, UI works so I think, Yes :D

**Pull request merged**